### PR TITLE
Fix database XMaterial loading

### LINKED ISSUE #298: Cannot get enum value of 'GRASS_PATH'
**Describe the bug**
plugin no worky with db )

**To Reproduce**
1. Use any version above 3.0.3
2. Load the db file attached
3. See error

**Expected behavior**
for plugin to work

** Server Version and Plugin Version**
Versions 3.0.3+, Minecraft 1.16.5

**Screenshots**

**Logs**
If there are error messages in your console, please send them via [This](https://paste.gg/ )
https://paste.gg/p/anonymous/09568d2eae4e43129c7e457a6ee19d83

**Additional context**
Resetting the db file fixes this

### PR DESCRIPTION
**Pull request by xReveres**
Fixes #298 and other future XMaterial changes

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/database/types/XMaterialType.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        return null;
+    }
+
+    @Override
+	public Object javaToSqlArg(FieldType fieldType, Object obj) throws SQLException {
+        XMaterial material = (XMaterial) obj;
+        return super.javaToSqlArg(fieldType, material.toString());
+    }
+    
+}
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/database/types/XMaterialType.java:37]: The indentation in this class is something else
```suggestion
public class XMaterialType extends StringType {

    private static final XMaterialType instance = new XMaterialType();

    public static XMaterialType getSingleton() {
        return instance;
    }

    protected XMaterialType() {
        super(SqlType.STRING, new Class<?>[] { XMaterial.class });
    }

    @Override
    public Object sqlArgToJava(FieldType fieldType, Object sqlArg, int columnPos) throws SQLException {
        String value = (String) super.sqlArgToJava(fieldType, sqlArg, columnPos);
        Optional<XMaterial> material = XMaterial.matchXMaterial(value);
        return material.orElse(null);
    }

    @Override
    public Object javaToSqlArg(FieldType fieldType, Object object) throws SQLException {
        XMaterial material = (XMaterial) object;
        return super.javaToSqlArg(fieldType, material.toString());
    }
    
}

```

**Review by dlsf — CHANGES_REQUESTED** [review]
Also, I don't quite get the point of that singleton, is it required by OrmLite?

**Comment by xReveres** [comment]
I inspired from this
https://github.com/j256/ormlite-core/blob/master/src/main/java/com/j256/ormlite/field/types/SerializableType.java

**Comment by dlsf** [comment]
I see. Did you test if these changes actually work? OrmLite sometimes just ignores stuff that has been configured

**Comment by xReveres** [comment]
Yes I tested it 2 times, for example I manually changed in database from "DIRT_PATH" to "GRASS_PATH" and works fine

**Review by dlsf — APPROVED** [review]

**Pull request merged**